design/sales.md

```diff
@@ -45,3 +45,49 @@ the Reservations module, so that it survives node restarts.
     Datastore API:
       function put(key, value)
       function get(key): value
+
+Reserving storage space
+-----------------------
+
+When a request for storage is submitted on chain, the sales module decides
+whether or not it wants to act on it. If the requested duration and size match
+what the node wants to sell and the reward is sufficient, then the sales module
+will go through several steps to try and fill a slot in the request.
+
+First, it will select a slot from the request to fill. Then, it will reserve
+space in the Repo to ensure that it keeps enough space available to store the
+content. It will mark the reserved space as belonging to the slot. Next, it will
+download the content, calculate a storage proof, and submit the proof on chain.
+If any of these later steps fail, then the node should release the storage that
+it reserved earlier.
+
+When a slot was filled successfully, then its storage should not be released
+until the request ends.
+
+Releasing storage space
+-----------------------
+
+When a storage request ends, or an attempt to fill a slot fails, it is important
+to release the reserved space that belonged to the slot. To ensure that the
+right amount of space is released, and that it is only released once, we keep
+track of how much space is reserved in the Repo for a slot.
+
+Releasing storage space goes in three steps. First, we look up the amount of
```

Comment:
We should probably use the correct terminology to avoid confusion. 

The repo has 3 types of storage:

- Available - unused and unreserved space, the repo is allowed to either physically use this bytes by writing blocks on disk or mark them as reserved
- Reserved - bytes that are not available, but aren't physically occupied on disk. This is usually used to mark space as "reserved" for rent/sell
- Used - physically used bytes on drive

The repo exposes these parameters:

```nim
    quotaMaxBytes*: uint          # maximum available bytes
    quotaUsedBytes*: uint         # bytes used by the repo
    quotaReservedBytes*: uint     # bytes reserved by the repo
```

Furthermore, the `totalUsed` prop exposes the total number reserved+used bytes:

```nim
func totalUsed*(self: RepoStore): uint =
  (self.quotaUsedBytes + self.quotaReservedBytes)
```

When writing block on disk, the `totalUsed + data.len <= quotaMaxBytes` is used to determined if there is enough room available.

When talking about the repo then, we should say - `reserved/released` when space is set as reserved but not physically used and `used/free` when space is physically used or freed.